src/interfaces/interfaces.ts

```diff
@@ -90,7 +92,7 @@ namespace interfaces {
     factory: FactoryTypeFunction | null
   };
 
-  export type Provider<T> = (...args: any[]) => (((...args: any[]) => Promise<T>) | Promise<T>);
+  export type Provider<T> = (...args: never[]) => (((...args: never[]) => Promise<T>) | Promise<T>);
```

Comment:
I would highly recommend leaving this type as `any`. `never` is not really a good candidate for this use case.

In my mind, `any` is acceptable for defining function args with unknown arity.